lib/python/picongpu/input/parameters.py

```diff
@@ -240,8 +275,8 @@ def set_value(self, x):
         A float containing the internal 'value' attribute after adjustment.
         """
 
-        self.value = self.base ** x
-        return self.value
+        self.value = self.dtype(self.base ** x)
+        return self.formatter(self.value)
 
     def get_value(self):
```

Comment:
Before calling ```log10(self.value)``` we need to cast the value back to float, because in case we use ```self.dtype = int```, I got the error `AttributeError: 'int' object has no attribute 'log10'`